app/api/src/crud/crud_layer.py

```diff
@@ -1,9 +1,69 @@
+from typing import Any, List
+from sqlalchemy.ext.asyncio import AsyncSession
 from .base import CRUDBase
 from src.db.models.layer import Layer
+from .crud_content import content as crud_content
+from src.schemas.layer import LayerRead
+from src.db.models.content import Content
+
+
+from sqlalchemy import select
 
 
 class CRUDLayer(CRUDBase):
-    pass
+    async def create(self, db, *, obj_in):
+        content = await crud_content.create(db, obj_in=obj_in, content_type="layer")
+        obj_in = Layer(**obj_in.dict())
+        obj_in.content_id = content.id
+        layer = await super().create(db, obj_in=obj_in)
+        layer = layer.__dict__
+        layer.update(content.__dict__)
+        layer = LayerRead(**layer)
+        return layer
+
+    async def get(self, db, *, id):
+        statement = select([Layer, Content]).join(Content).where(Layer.content_id == str(id))
+        layer = await db.execute(statement)
+        layer = layer.fetchone()
+        layer_dict = layer.Layer.__dict__
+        layer_dict.update(layer.Content.__dict__)
+        return layer_dict
+
+    async def update(self, db, *, db_obj, obj_in):
+        content_db_obj = await crud_content.get(db, id=str(obj_in.content_id))
+        content_in = Content(**obj_in.dict())
+        content = await crud_content.update(db, db_obj=content_db_obj, obj_in=content_in)
+        db_obj = select(Layer).where(Layer.content_id == str(obj_in.content_id))
+        db_obj = await db.execute(db_obj)
+        db_obj = db_obj.fetchone()[0]
+        obj_in = Layer(**obj_in.dict())
+        layer = await super().update(db, db_obj=db_obj, obj_in=obj_in)
+        layer = layer.__dict__
+        layer.update(content.__dict__)
+        return layer
+
+    async def get_multi(
+        self,
+        db: AsyncSession,
+        *,
```

Comment:
Recently we did some tests with the fastapi-pagination library. Majk tested it in the auth project. I can send a link: https://github.com/goat-community/auth/blob/authz/app/api/src/endpoints/v1/regions.py

Maybe we can make this the standard also in the GOAT repo.